src/components/SearchBar.jsx

```diff
@@ -0,0 +1,72 @@
+// LIBRARY IMPORTS
+import React, { useState, useRef } from 'react';
+import { MagnifyingGlassIcon, XMarkIcon } from '@heroicons/react/24/solid';
+
+// LOCAL IMPORTS
+
+export default function SearchBar({ data, setData, setClearedSearch, fullDataSet }) {
+  // SET STATES
+  const [searchSubmitted, setSearchSubmitted] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [searchSuccessful, setSearchSuccessful] = useState(false);
+  const searchTermRef = useRef('');
+
```

Comment:
These states on lines 9-11 all help determine whether a search has been submitted or not, and whether or not it was successful. Then below, certain text/buttons/functionality render based on these states. `useRef` also makes an appearance here as a way to save the string of text that was submitted in the search and then render that string in a text block under the search bar when there's a successful search so users know which search results they're viewing.